cdc/generation.cc

```diff
@@ -414,13 +447,17 @@ generation_service::~generation_service() {
     assert(_stopped);
 }
 
-future<> generation_service::after_join() {
+future<> generation_service::after_join(std::optional<db_clock::time_point>&& startup_gen_ts) {
```

Comment:
See [my comment](https://github.com/scylladb/scylla/pull/8101/files#r576991466) about the move constructor of `std::optional`. I think this function can be changed to take an optional timestamp by value - using rvalue reference here is an overkill.

Reply:
the rvalue reference serves as a hint/documentation. It tells the caller: pass me the ownership of the generation timestamp (and don't ever touch it again yourself).

Reply:
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rf-consume

Reply:
Ok. This function will be called once, so performance-wise it doesn't matter.